### PR TITLE
Install dvo on rhtap clusters

### PR DESCRIPTION
**Pull request by bamachrn**
* This PR installs Deployment-Validation-Operator(DVO) in all RHTAP clusters.
* DVO works as a second level of checking on the installed CRs as kube-lint in the CI.
* This is part of the following best practices effort https://issues.redhat.com/browse/RHTAPSRE-93

### CODE REVIEW

**Comment by hugares** [comment]
I tested this change in my cluster and the subscription was degraded:
```
conditions:
    - lastTransitionTime: '2023-09-15T13:19:12Z'
      message: all available catalogsources are healthy
      reason: AllCatalogSourcesHealthy
      status: 'False'
      type: CatalogSourcesUnhealthy
    - message: >-
        failed to populate resolver cache from source
        operatorgroup-unavailable/deployment-validation-operator: found 0
        operatorgroups in namespace deployment-validation-operator: expected 1
      reason: ErrorPreventedResolution
      status: 'True'
      type: ResolutionFailed
```
The operator group is missing. I created manually the following operatorgroup and vdo installation went through:
```
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: deployment-validation-operator
  namespace: deployment-validation-operator
spec:
  targetNamespaces:
    - deployment-validation-operator

```

**Comment by hugares** [comment]
> @hugares thanks for the review, yes I was tesing a way if installing this using ArgoCD. now it works locally. can I request for another review please?

Sure, let me try again on my side with latest revision.

**Comment by hugares** [comment]
I tested this change in my cluster and it works. One question:  How do we fetch/view the DVO results, i.e. the issues detected?

**Comment by bamachrn** [comment]
> I tested this change in my cluster and it works. One question: How do we fetch/view the DVO results, i.e. the issues detected?

These are exported by the prometheus metrics exporters. So the installation creates a metrics-exporter service, this allows the results/alerts to be pulled by the prometheus/grafana. My next step after this PR is to add these to the alerts so that we know if any issues detected.

**Review by hugares — APPROVED** [review]
/lgtm
